### ISSUE TITLE
Bug in B030: False positive when adding tuples of exception classes

### ISSUE THREAD

**Issue by autinerd** [astral-sh/ruff#10426]
Hello :slightly_smiling_face: 

When enabling rule B030, these cases are being flagged as violated:

```python
try:
    ...
except (ValueError,TypeError) + (EOFError,ArithmeticError):
    ...
```

```python
BASE_EXCEPTIONS = (ValueError,TypeError)

try:
    ...
except BASE_EXCEPTIONS + (EOFError,ArithmeticError):
    ...
```

Ruff version 0.3.2

**Comment by ottaviohartman** [comment]
I can try to tackle this one!

One idea is to check for `BinOp`'s on tuples and flatten -- is this the right direction?

**Comment by autinerd** [comment]
Sounds good to me!

Using `RUF005` to replace tuple concatenations with unpacking eliminates this case of course, because tuple unpacking is detected correctly.
